code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -352,6 +352,10 @@
 		M.visible_message(span_notice("[M] hugs [src] to make [p_them()] feel better!"), \
 					span_notice("You hug [src] to make [p_them()] feel better!"))
 
+		if(IS_VAMPIRE(M) && src.client)
+			var/datum/antagonist/vampire/vampdatum = IS_VAMPIRE(M)
+			SEND_SIGNAL(vampdatum, COMSIG_VAMPIRE_TRACK_HUMANITY_GAIN, HUMANITY_HUGGING_TYPE, src)
+
```

Comment:
This is bad code, find a better way to do this that doesn't involve non-generic checks inside a highly generic function.